src/storage/page_buffer.c

```diff
@@ -15778,3 +15846,259 @@ pgbuf_init_temp_page_lsa (FILEIO_PAGE * io_page, PGLENGTH page_size)
+	{
+	  pgbuf_Pool.show_status.now.victim_candidate_pages++;
+	}
+
+      /* count temporary and permanent pages */
+      if (pgbuf_is_temporary_volume (vpid.volid) == true)
+	{
+	  pgbuf_Pool.show_status.now.num_temp_pages++;
+
+	  assert ((page_type == PAGE_UNKNOWN) ||	/* dealloc pages, we don't know page type */
+		  (page_type == PAGE_AREA) || (page_type == PAGE_QRESULT) ||	/* temporary page type */
+		  (page_type == PAGE_EHASH) || (page_type == PAGE_VOLHEADER)	/* It can be temporary or permanent pages */
+		  || (page_type == PAGE_VOLBITMAP) || (page_type == PAGE_FTAB));	/* It can be temporary or permanent pages */
+	}
+      else
+	{
+	  switch (page_type)
+	    {
+	    case PAGE_BTREE:
+	      pgbuf_Pool.show_status.now.num_index_pages++;
+	      break;
+	    case PAGE_OVERFLOW:
+	    case PAGE_HEAP:
+	      pgbuf_Pool.show_status.now.num_data_pages++;
+	      break;
+	    case PAGE_CATALOG:
+	    case PAGE_VOLBITMAP:
+	    case PAGE_VOLHEADER:
+	    case PAGE_FTAB:
+	    case PAGE_EHASH:
+	    case PAGE_VACUUM_DATA:
+	    case PAGE_DROPPED_FILES:
+	      pgbuf_Pool.show_status.now.num_system_pages++;
+	      break;
+	    default:
+	      /* dealloc pages, we don't know page type */
+	      assert (page_type == PAGE_UNKNOWN);
+	      break;
+	    }
+	}
+    }
+}
+
+/*
+ * pgbuf_start_scan () - start scan function for show page buffer status
+ *   return: NO_ERROR, or ER_code
+ *
+ *   thread_p(in):
+ *   type (in):
+ *   arg_values(in):
+ *   arg_cnt(in):
+ *   ptr(in/out):
+ */
+int
+pgbuf_start_scan (THREAD_ENTRY * thread_p, int type, DB_VALUE ** arg_values, int arg_cnt, void **ptr)
+{
+  SHOWSTMT_ARRAY_CONTEXT *ctx = NULL;
+  const int num_cols = 19;
+  time_t cur_time;
+  int idx;
+  int error = NO_ERROR;
+  DB_VALUE *vals = NULL, db_val;
+  unsigned long long delta, hit_delta, request_delta;
+  double time_delta;
+  double hit_rate;
+  DB_DATA_STATUS data_status;
+
+  *ptr = NULL;
+
+#if defined(SERVER_MODE)
+  (void) pthread_mutex_lock (&pgbuf_Pool.show_status_mutex);
+#endif
+
+  pgbuf_scan_bcb_table ();
+
+  ctx = showstmt_alloc_array_context (thread_p, 1, num_cols);
+  if (ctx == NULL)
+    {
+      error = er_errid ();
+      return error;
+    }
+
+  vals = showstmt_alloc_tuple_in_context (thread_p, ctx);
+  if (vals == NULL)
+    {
+      error = er_errid ();
+      goto exit_on_error;
+    }
+
+  cur_time = time (NULL);
+
+  time_delta = difftime (cur_time, pgbuf_Pool.show_status.old.print_out_time) + 0.0001;	// avoid dividing by 0
+
+  idx = 0;
+
+  hit_rate = (pgbuf_Pool.show_status.now.num_hit - pgbuf_Pool.show_status.old.num_hit) /
+    ((pgbuf_Pool.show_status.now.num_page_request - pgbuf_Pool.show_status.old.num_page_request) + 0.0000000000001);
+  hit_rate = hit_rate * 100;
+
+  db_make_double (&db_val, hit_rate);
+  db_value_domain_init (&vals[idx], DB_TYPE_NUMERIC, 13, 10);
+  error = numeric_db_value_coerce_to_num (&db_val, &vals[idx], &data_status);
```

Comment:
can you change three functions (db_make_double, db_value_domain_init, numeric_db_value_coerce_to_num) to db_name_numeric ?

Reply:
@mhoh3963 Do you mean the db_make_numeric() function? this function takes an unsigned char *(DB_C_NUMERIC) type as an argument, not a floating-point type. therefore, it is not considered suitable in this case.

Reply:
OK, i think as your comment that db_make_numeric() function is not suitable in this code.